README.md

```diff
@@ -1,7 +1,48 @@
 # Cluster Deployment Automation
-Automate deployment of clusters in different configurations. An example configuration below showcases the main functionality:
+Automate deployment of OpenShift clusters in different configurations.
+The main functionality is showcased below.  The distribution specific
+steps assume a RHEL/Fedora based distribution is used.  Also, the
+instructions below make the assumption that everything is run as root
+from the repository root directory.
 
+## Generate a ssh key
+NOTE: starting with Fedora 33 RSA keys are considered not secure enough; use
+ed25519 instead.
+
+```
+ssh-keygen -t ed25519 -a 64 -N '' -f ~/.ssh/id_ed25519
+```
+
+## Install required software
+```
+dnf install -y libvirt podman qemu virt-install 
+systemctl enable libvirtd
+```
+
+## Configure qemu user an group to be root
+```
+sed -e 's/#\(user\|group\) = ".*"$/\1 = "root"/' -i /etc/libvirt/qemu.conf
+systemctl restart libvirtd
+```
+
+## Install required Python packages
+```
+pip install -r requirements.txt
 ```
+
+## Install the latest 4.x openshift-client
+```
+pushd /tmp/
+wget https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/openshift-client-linux.tar.gz
+tar xf openshift-client-linux.tar.gz oc
+rm openshift-client-linux.tar.gz
+mv oc /usr/bin/oc
+popd
+```
```

Comment:
The correct version of oc is now pulled in automatically. This isn't needed anymore.

Reply:
Ah, I missed https://github.com/bn222/cluster-deployment-automation/commit/6c147b34fe0fcaaa55dbef2a5b62f58b0d9b9598
Do you think it's useful to keep as reference though so that people that want to try this on fedora don't end up using old `oc` binaries (Fedora ships version 3.x which is very old)?

I can rename the section to something like `Install the latest 4.x openshift-client (optional)`.

Reply:
Yes, but it let's put it in a separate section to delineate between what's really a dep and what's good to know

Reply:
What do you think of this incremental change?

````
diff --git a/README.md b/README.md
index 3cd2d3f..dc2a230 100644
--- a/README.md
+++ b/README.md
@@ -25,12 +25,18 @@ sed -e 's/#\(user\|group\) = ".*"$/\1 = "root"/' -i /etc/libvirt/qemu.conf
 systemctl restart libvirtd
 ```
 
-## Install required Python packages
+## Install required Python packages (in a virtualenv)
 ```
+python -m venv ocp-venv
+source ocp-venv/bin/activate
 pip install -r requirements.txt
+deactivate
 ```
 
-## Install the latest 4.x openshift-client
+## Install the latest 4.x openshift-client (optional)
+The scripts don't rely on the OpenShift Client (oc) being installed locally.
+For a better user experience it might still be a good idea to install the
+latest 4.x version:
 ```
 pushd /tmp/
 wget https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/openshift-client-linux.tar.gz
@@ -104,7 +110,9 @@ EOF
 
 ## Start the installation
 ```
+source ocp-venv/bin/activate
 python main.py cluster.yaml
+deactivate
 ```
 
 ## Set the kubeconfig after a successful installation
````

Reply:
sounds good